app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseDatabase.kt

```diff
@@ -297,6 +300,94 @@ class FirebaseDatabase(activity: Activity) : Database {
     override suspend fun unfollow(follower: String, followee: String) {
         doFollow(follower, followee, follow = false)
     }
-}
 
+    /**
+     * Returns a list of users that liked a given challenge
+     * @param cid the challenge id
+     * @return a list of users that liked the challenge
+     */
+    internal fun getLikesOf(cid: String): Task<Map<String, Boolean>> {
+        return dbChallengeRef.child(cid).child("likes").get()
+            .thenMap { snapshot ->
+                snapshot.toMap<Boolean>().withDefault { false }
+            }
+    }
+
+    /**
+     * Function updating the firebase making user like/dislike a given challenge
+     *
+     * @note To ensure easier data querying, the data must be synchronized at 3 places in the database :
+     *      - In the user's like list, the challenge should be added/removed,
+     *      - The challenge's counter should be incremented/decremented,
+     *      - The user should be added/removed to the challenge's liker list.
+     */
+    private suspend fun doLike(uid: String, cid: String, like: Boolean = true) {
+        // TODO Writes are not made atomically and should be batched instead
+        //      See https://github.com/SDP-GeoHunt/geo-hunt/issues/88#issue-1647852411
+
+        val userLikesRef = dbUserRef.child(uid).child("likes")
+        val challengeLikesRef = dbChallengeRef.child(cid).child("likes")
```

Comment:
```suggestion
        val challengeLikesRef = dbChallengeRef.child(cid).child("likedBy")
```

Maybe update this field for a more "understandable" one